BrainPortal/lib/boutiques_support.rb

```diff
@@ -89,18 +90,26 @@ def self.new_from_string(text)
 
     def self.new_from_file(path)
       obj = self.new_from_string(File.read(path))
-      obj.from_file = path
+      obj.from_file     = path
+      obj.mtime_of_file = File.mtime(path)
       obj
     end
 
+    def reload_if_file_timestamp_changed()
+      filepath = self.from_file
+      return self if filepath.blank? || File.mtime(filepath) == self.mtime_of_file
```

Comment:
You cannot truste that you can compare timestamps with `==` ; it is posisble the internal time and the filesystem times do not have the same granularity. Instead compare with something like `(time1 - time2).abs < 1.second`

Reply:
not sure I understand how this can happen. Do you mean ruby rounds up the system file's timestamp differently time from time?

Reply:
It worked on my machine, and in worth case it is just a refresh, so I did not pay much attention to that.

Reply:
OK keep the == comparison.

Reply:
well I added the comparison,seems work fine. Indeed time may be tricky better be on safe side here